recipes/wasmtime/all/conanfile.py

```diff
@@ -0,0 +1,106 @@
+    settings = "os", "compiler", "arch"
+    options = { "shared": [True, False] }
+    default_options = { 'shared': False }
+    no_copy_source = True
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "apple-clang": "9.4",
+            "clang": "3.3",
+            "gcc": "5.1"
+        }
+
+    @property
+    def _sources_key(self):
+        if self.settings.compiler == "Visual Studio":
+            return "Windows"
+        elif self.settings.os == "Windows" and self.settings.compiler == "gcc":
+            return "MinGW"
+        return str(self.settings.os)
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.runtime
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        compiler = self.settings.compiler
+        min_version = self._minimum_compilers_version[str(compiler)]
+        try:
+            if tools.Version(compiler.version) < min_version:
+                msg = (
+                    "{} requires C{} features which are not supported by compiler {} {} !!"
+                ).format(self.name, self._minimum_cpp_standard, compiler, compiler.version)
+                raise ConanInvalidConfiguration(msg)
+        except KeyError:
+            msg = (
+                "{} recipe lacks information about the {} compiler, "
+                "support for the required C{} features is assumed"
+            ).format(self.name, compiler, self._minimum_cpp_standard)
+            self.output.warn(msg)
+
+        try:
+            self.conan_data["sources"][self.version][self._sources_key][str(self.settings.arch)]
+        except KeyError:
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
+
+        if (self.settings.compiler, self.settings.os) == ("gcc", "Windows") and self.options.shared:
+            # https://github.com/bytecodealliance/wasmtime/issues/3168
+            raise ConanInvalidConfiguration("Shared mingw is currently not possible")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][self._sources_key][str(self.settings.arch)], destination=self.source_folder, strip_root=True)
+
+    def package(self):
+        shutil.copytree(os.path.join(self.source_folder, "include"),
+                        os.path.join(self.package_folder, "include"))
+
+        self.copy('*.dll', dst='bin', keep_path=False)
+        self.copy('*.lib', dst='lib', keep_path=False)
+        self.copy('*.so', dst='lib', keep_path=False)
+        self.copy('*.dylib', dst='lib', keep_path=False)
+        self.copy('*.a', dst='lib', keep_path=False)
+
```

Comment:
Please only copy the static libraries in a static `wasmtime`, and only copy the shared libraries in a shared `wasmtime`.

Reply:
Still do not understand why it is important ...

For detecting the issues with copied wrong library we have CI

Reply:
The `self.options.shared` option declares what type of library the package provides.
Right now, it provides both.
So there is no ambiguity what type of library is linked against.

Reply:
Done